src/main/java/com/microsoft/jenkins/azuread/integrations/casc/AzureAdMatrixAuthorizationStrategyConfigurator.java

```diff
@@ -0,0 +1,110 @@
+        BaseConfigurator<AzureAdMatrixAuthorizationStrategy> {
+    @NonNull
+    @Override
+    public String getName() {
+        return "azureAdMatrix";
+    }
+
+    @Override
+    public Class<AzureAdMatrixAuthorizationStrategy> getTarget() {
+        return AzureAdMatrixAuthorizationStrategy.class;
+    }
+
+
+    @NonNull
+    @Override
+    public Class getImplementedAPI() {
+        return AuthorizationStrategy.class;
+    }
+
+    @Override
+    @NonNull
+    public Set<Attribute<AzureAdMatrixAuthorizationStrategy, ?>> describe() {
+        return new HashSet<>(Arrays.asList(
+                new MultivaluedAttribute<AzureAdMatrixAuthorizationStrategy, String>("permissions", String.class)
+                        .getter(AzureAdMatrixAuthorizationStrategyConfigurator::getPermissions)
+                        .setter(AzureAdMatrixAuthorizationStrategyConfigurator::setPermissions),
+
+                // support old style configuration options
+                new MultivaluedAttribute<AzureAdMatrixAuthorizationStrategy, String>("grantedPermissions", String.class)
+                        .getter(unused -> null)
+                        .setter(AzureAdMatrixAuthorizationStrategyConfigurator::setPermissionsDeprecated)
+        ));
+    }
+
+    @Override
+    protected AzureAdMatrixAuthorizationStrategy instance(Mapping mapping, ConfigurationContext context) {
+        return new AzureAdMatrixAuthorizationStrategy();
+    }
+
+    @Override
+    public CNode describe(AzureAdMatrixAuthorizationStrategy instance, ConfigurationContext context) throws Exception {
+        return compare(instance, new AzureAdMatrixAuthorizationStrategy(), context);
+    }
+
+
+    /**
+     * Extract container's permissions as a List of "PERMISSION:sid".
+     */
+    public static Collection<String> getPermissions(AuthorizationContainer container) {
+        return container.getGrantedPermissions().entrySet().stream()
+                .flatMap(e -> e.getValue().stream()
+                        .map(v -> e.getKey().group.getId() + "/" + e.getKey().name + ":" + v))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Configure container's permissions from a List of "PERMISSION:sid".
+     */
+    public static void setPermissions(AuthorizationContainer container, Collection<String> permissions) {
+        permissions.forEach(p -> {
+            final int i = p.indexOf(':');
+            final Permission permission = PermissionFinder.findPermission(p.substring(0, i));
+            if (permission == null) {
+                throw new IllegalStateException(String.format("Cannot find permission for %s.", p));
+            }
+            container.add(permission, p.substring(i + 1));
+        });
+    }
+
+    /**
+     * Like {@link #setPermissions(AuthorizationContainer, Collection)} but logs a deprecation warning.
+     */
+    public static void setPermissionsDeprecated(AuthorizationContainer container, Collection<String> permissions) {
```

Comment:
remove, not needed, matrix-auth was maintaining compat from configuration-as-code-support configurator. This plugin never had support previously